logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/asset/GetMessageAssetUseCase.kt

```diff
@@ -116,7 +118,15 @@ internal class GetMessageAssetUseCaseImpl(
                         ).fold({
                             kaliumLogger.e("There was an error downloading asset with id => ${assetMetadata.assetKey.obfuscateId()}")
                             // This should be called if there is an issue while downloading the asset
-                            updateAssetMessageDownloadStatus(Message.DownloadStatus.FAILED_DOWNLOAD, conversationId, messageId)
+                            if (it is NetworkFailure.ServerMiscommunication &&
+                                it.kaliumException is KaliumException.InvalidRequestError
+                                && it.kaliumException.isNotFound()
```

Comment:
maybe in this case we need to handle the exact error label from the backend 
because this code does not differentiate between  the image is 404 or the end point is 404

Reply:
I will ask backend what kind of label do they return

Reply:
NVM, the server error label is too generic